### PR TITLE
Add promqlsmith fuzzer

### PR DESCRIPTION
**Pull request by yeya24**
This PR uses the new PromQLsmith query fuzzer to replace our old fuzzing test.

Fixes https://github.com/thanos-community/promql-engine/issues/209.

Note that the fuzzer already found some bugs at https://github.com/thanos-community/promql-engine/issues/200 but currently we don't have a way to tell the fuzzer to avoid generating such kind of queries. In this case, the CI will fail.

I am also wondering if we should have a dedicated workflow for running the fuzz test. And currently we don't have a way to collect queries bugs from our test run.

### CODE REVIEW

File: engine/enginefuzz_test.go [hunk 1]
```diff
@@ -32,108 +37,141 @@ func FuzzEngineQueryRangeMatrixFunctions(f *testing.F) {
...
+		ps := promqlsmith.New(rnd, seriesSet, psOpts...)
+
+		newEngine := engine.New(engine.Opts{EngineOpts: opts, DisableFallback: true})
+
+		var (
+			q1    promql.Query
+			query string
+		)
+		// Since we disabled fallback, keep trying until we find a query
+		// that can be natively execute by the engine.
```
Comment (fpetkovski) [inline engine/enginefuzz_test.go:78]:
```suggestion
		// that can be natively executed by the engine.
```

File: engine/new.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
+goos: darwin
```
Comment (fpetkovski) [inline engine/new.txt:1]: Is this supposed to be in the commit?

**Review by fpetkovski — COMMENTED** [review]
Thanks, this is a great addition.

File: engine/new.txt (hunk 2 above)
Comment (yeya24) [reply engine/new.txt:1]: Ah, sorry I should exclude this.

**Comment by yeya24** [comment]
Fixed @fpetkovski Please take another look.

**Review by fpetkovski — APPROVED** [review]
Thanks, lgtm

**Pull request merged**